src/beginner/dict_vs_defaultdict/dict_vs_defaultdict.py

```diff
@@ -0,0 +1,65 @@
+from collections import defaultdict
+from typing import Optional
+
+
+def get_value_from_key_with_square_brackets(
+    my_dict: dict[str, str],
+    key: str,
+) -> str:
+    """
+    Get value from a dict using square brackets.
+
+    Parameters:
+        my_dict (dict[str, str]): dict to find value.
+        key (str): key to find in the dict.
+
+    Returns:
+        value (str): value of the dict for the key provided.
+
+    Raises:
+        KeyError: If key is not in the dict.
+    """
+    return my_dict[key]
+
+
+def get_value_from_key_with_get(
+    my_dict: dict[str, str],
+    key: str,
+    default: str = None,
+) -> Optional[str]:
+    """
+    Get value from a dict using get method.
+    Return default if key is not in the dict.
+
+    Parameters:
+        my_dict (dict[str, str]): dict to find value.
+        key (str): key to find in the dict.
+        default (str): default value to return if key is not in the
+            dict.
+
+    Returns:
+        value (str, None): value of the dict for the key provided.
+    """
+    return my_dict.get(key, default)
+
+
+def get_value_from_defaultdict(
+    my_dict: dict[str, str],
+    key: str,
+    default: str = None,
```

Comment:
defaultdict has a parameter, default_factory, which is not an string. It is a function that return a value in case of the given value is not in the dict.
if the default_factory is not provided, the dictionary will return a KeyError exception.

Reply:
well spotted